src/metpy/calc/thermo.py

```diff
@@ -252,6 +298,10 @@ def moist_lapse(pressure, temperature, reference_pressure=None):
 
     This equation comes from [Bakhshaii2013]_.
 
+    Only functions on 1D profiles (not higher-dimension vertical cross sections or grids).
+    Since this function returns scalar values when given a profile, this will return Pint
```

Comment:
Good point! I mistakenly went off of the docstring (which implied singular temperature as a return value), rather than the actual implementation in the function. However, unlike `dry_lapse`, it can still only function on 1D profiles because of how indexing is done. It could probably be refactored to vectorize better, but that is a concern for another time.